DEPRECATION.md

```diff
@@ -0,0 +1,91 @@
+# Deprecation Policy Proposal
+
+As AllenNLP matures we will inevitably need to deprecate old models, interfaces and features. This document provides guidelines that attempt to minimize churn. Reducing churn helps users build next generation models with tools they've already mastered.
+
+### Terminology note
+
+We _deprecate_ some component to communicate that a preferred alternative exists. A _deprecated_ component does not necessarily have to be _removed_ from the library.
+
+### Goals
+
+1. Promote backward compatibility where not onerous.
+2. Describe code-level and release process for both deprecation and removal.
+
+### Non-goals
+
+1. Eliminate all deprecation and removal.
+2. Legislate. Good judgment by AllenNLP developers is much preferred.
+
+## Decision Flow
+
+Say you've found some old code that needs some care. You have a fix in mind, maybe already a PR. Great, let's get it merged! But let's give some thought to backward compatibility along the way.
+
+1. Determine whether backward compatibility is even an issue.
+
+  * __Likely safe__
+
+    * New code
+    * Internal code _TODO: How is this determined?_
+    * Implementation change only
+
+  * __Care required__
+
+    * File/class/function names
+      * Including names passed to `Registrable.register`!
+    * Function signatures
+    * Jsonnet config
```

Comment:
I'm not sure what this means.  Do you mean changing an existing file in `training_config`?  Or something else?